libsolidity/analysis/SyntaxChecker.cpp

```diff
@@ -138,9 +138,25 @@ void SyntaxChecker::endVisit(ModifierDefinition const& _modifier)
 	m_placeholderFound = false;
 }
 
-bool SyntaxChecker::visit(WhileStatement const&)
+void SyntaxChecker::checkSingleStatementVariableDeclaration(ASTNode const* _statement)
```

Comment:
The argument here could probably actually be of type ``Statement const*``, since it's only used for types that derive from ``Statement``. Since there is another cast below anyways, ``ASTNode const*`` should be fine as well, though (and may even have advantages).

Reply:
Yea, we could have the cast via the parameter itself and just check for not null in the function. Though that might become less clear

Reply:
My point was that in all current use cases of this function the node that is passed in already *is* a ``Statement`` (even ``Block`` derives from ``Statement``), so the argument type could be changed to ``Statement`` instead of ``ASTNode`` without having to add any additional casts or changing anything else... but it really doesn't matter too much... and we might even want to call this with a node that is not necessarily a ``Statement`` in the future, so yeah, we can just keep it.

Reply:
Yea, that was my original thought.